app/src/main/java/com/teamwss/websoso/ui/storage/StorageActivity.kt

```diff
@@ -19,26 +22,67 @@ class StorageActivity : BaseActivity<ActivityStorageBinding>(R.layout.activity_s
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        bindViewModel()
         setupViewPagerAndTabLayout()
+        onSortTypeButtonClick()
         onBackButtonClick()
     }
 
+    private fun bindViewModel() {
+        binding.lifecycleOwner = this
+        binding.storageViewModel = storageViewModel
+    }
+
     private fun setupViewPagerAndTabLayout() {
-        val pagerAdapter =
-            StorageViewPagerAdapter(storageViewModel.novelsMap, ::navigateToExploreFragment)
+        setupViewPager()
+        setupTabLayoutWithViewPager()
+        setupObserver()
+    }
+
+    private fun setupViewPager() {
+        val pagerAdapter = StorageViewPagerAdapter(
+            novels = emptyList(),
+            navigateToExplore = ::navigateToExploreFragment,
+            novelClickListener = ::navigateToNovelDetail,
+        )
         binding.vpStorage.adapter = pagerAdapter
-        bindTabLayoutWithViewPager()
     }
 
-    private fun bindTabLayoutWithViewPager() {
-        TabLayoutMediator(binding.tlStorage, binding.vpStorage) { tab, position ->
-            tab.text = StorageTab.fromPosition(position).title
-        }.attach()
+    private fun setupTabLayoutWithViewPager() {
+        binding.vpStorage.adapter?.let {
+            TabLayoutMediator(binding.tlStorage, binding.vpStorage) { tab, position ->
+                tab.text = StorageTab.fromPosition(position).title
+            }.attach()
+
+            binding.tlStorage.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+                override fun onTabSelected(tab: TabLayout.Tab?) {
+                    tab?.let {
+                        onSortTabSelected(it.position)
+                    }
+                }
+                override fun onTabUnselected(tab: TabLayout.Tab?) {}
+                override fun onTabReselected(tab: TabLayout.Tab?) {}
```

Comment:
c: 사용하지 않는 부분이라면 요런식으로 빼도 좋을 것 같아요
```suggestion
                override fun onTabUnselected(tab: TabLayout.Tab?) = Unit
                override fun onTabReselected(tab: TabLayout.Tab?) = Unit
```

Reply:
감사함둥 수정하겠습니다!